arelle/plugin/validate/DBA/rules/fr.py

```diff
@@ -797,6 +797,59 @@ def rule_fr81(
         )
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+)
+def rule_fr89(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    DBA.FR89: If ClassOfReportingEntity is one of the following values:
+
+    Regnskabsklasse C, mellemstor virksomhed // Reporting class C, medium-size enterprise
+    Regnskabsklasse C, stor virksomhed // Reporting class C, large enterprise
+    Regnskabsklasse D // Reporting class D
+    Then TypeOfAuditorAssistance should be: Revisionspåtegning // Auditor's report on audited financial statements
+    """
+    modelXbrl = val.modelXbrl
+    facts = modelXbrl.factsByQname.get(pluginData.classOfReportingEntityQn)
+    auditorFacts = modelXbrl.factsByQname.get(pluginData.typeOfAuditorAssistanceQn)
+    if auditorFacts is not None:
+        for auditorFact in auditorFacts:
+            if auditorFact.xValid >= VALID and auditorFact.xValue in [
+                pluginData.auditedFinancialStatementsDanish,
+                pluginData.auditedFinancialStatementsEnglish
+            ]:
+                return
+    classFacts = []
+    if facts is not None:
+        for fact in facts:
+            if fact.xValid >= VALID:
+                if fact.xValue in [
+                    pluginData.reportingClassCLargeDanish,
+                    pluginData.reportingClassCLargeEnglish,
+                    pluginData.reportingClassCMediumDanish,
+                    pluginData.reportingClassCMediumEnglish,
+                    pluginData.reportingClassDDanish,
+                    pluginData.reportingClassDEnglish,
+                ]:
+                    classFacts.append(fact)
+        if len(classFacts) > 0:
```

Comment:
I believe that this logic is wrong. This should be checking to see if `classOfReportingEntityQn` is one of the specified values and if it is then it should look at the value of `typeOfAuditorAssistanceQn`. If `typeOfAuditorAssistanceQn` is not in the specified values then we fire an error.